### PR TITLE
Remove share providers and fix various share bugs

### PR DESCRIPTION
**Pull request by cdrini**
I started by trying to find a way to specify a direct share url from bookreader, but couldn't quite find a way to make that happen in a reasonable fashion. But did find some nice cleanups to do while investigating the code!

- Remove ShareProvider abstraction ; it was making things kinda convoluted
- Use `URLSearchParams` everywhere; fixed various url encoding issues
- Fix Tumblr share link incorrect
- Fix click region of share buttons not full-width (could only click on the _word_ Twitter, not the whole row).

### CODE REVIEW

File: test/iaux-sharing-options.test.ts [hunk 1]
```diff
@@ -38,23 +38,13 @@ describe('<iaux-in-share-panel>', () => {
...
-      const button =
-        el.shadowRoot && el.shadowRoot.querySelector(`a.${option.class}`);
+    el.sharingOptions.forEach((option, i) => {
+      const button = el.shadowRoot?.querySelectorAll('a')[i];
       expect(button).to.exist;
       expect(button?.getAttribute('href')).to.equal(option.url);
     });
   });
 
-  it('toggles visibility of embed options', async () => {
```
Comment (iisa) [inline test/iaux-sharing-options.test.ts:49]: hi, did we not want this?

File: src/menus/share-panel.ts [hunk 2]
```diff
@@ -66,55 +67,74 @@ export class IauxSharingOptions extends LitElement {
...
+    if (this.fileSubPrefix) {
+      shareUrl += `/${this.fileSubPrefix}`;
+    }
+    const shareBlurb = [
+      this.description,
+      this.creator,
+      'Free Download, Borrow, and Streaming',
+      'Internet Archive',
+    ]
+      .filter(Boolean)
```
Comment (iisa) [inline src/menus/share-panel.ts:80]: q: why do we need to filter? should we have defaults for the desc, creator?

File: test/iaux-sharing-options.test.ts [hunk 3]
```diff
@@ -38,23 +38,13 @@ describe('<iaux-in-share-panel>', () => {
...
-    const el = (await fixture(container())) as IauxSharingOptions;
-
-    el.toggleEmbedOptions(new Event('click'));
```
Comment (iisa) [inline test/iaux-sharing-options.test.ts:52]: we can probably do...

```suggestion
  el.shadowRoot.querySelector('embed dropdown').click();
```

**Review by iisa — APPROVED** [review]
looks good, just had a question if we can keep toggle test.

File: test/iaux-sharing-options.test.ts (hunk 1 above)
Comment (cdrini) [reply test/iaux-sharing-options.test.ts:49]: I switched to using the html details tag, so was able to delete our code managing the visibility!

File: src/menus/share-panel.ts (hunk 2 above)
Comment (cdrini) [reply src/menus/share-panel.ts:80]: Sometimes things didn't have a creator was the main case.

**Pull request merged**